src/geometry.rs

```diff
@@ -29,6 +38,71 @@ impl Point {
     }
 }
 
+/*
+So we can write
+let a = Point(...);
+let b = Point(...);
+let v: Vector = a - &b;
+*/
+impl std::ops::Sub<&Point> for Point {
+    type Output = Vector;
+
+    fn sub(self, other: &Self) -> Vector {
+        Vector(self.0 - other.0, self.1 - other.1, self.2 - other.2)
+    }
+}
+
+/*
+So we can write
+let a = Point(...);
+let v = Vector(...);
+let b: Point = a + &v;
+*/
+impl std::ops::Add<&Vector> for Point {
+    type Output = Self;
```

Comment:
To implement `Add` trait, we need to define several types and functions. Exactly, we need to define `Output` and `add`.
In the scope of the trait, `Self` refers to the type of the struct, for which the trait is implemented. In our case, this is `Point`, so it would be equivalent.
I agree, that writing `type Output = Point;` would be more clear in this case.